### ISSUE TITLE
HAP-Client uniqueId is different when a device is removed then added, even though HomeKit thinks its the same device

### ISSUE THREAD

**Issue by NorthernMan54** [oznu/hap-client#15]
### Current Situation

service.uniqueId = createHash('sha256')
            .update(`${service.instance.username}${service.aid}${service.iid}${service.type}`)
            .digest('hex');

AID increments when a accessory is removed and added

### Proposed Change

This creates the same token for the same device when deleting and adding  - `${service.instance.name}${service.instance.username}${service.accessoryInformation.Manufacturer}${friendlyName}${service.uuid.slice(0, 8)}`

**Comment by bwp91** [comment]
@NorthernMan54 could we revisit this proposal again? it would be really helpful for the UI if accessories always had the same unique id.

**Comment by NorthernMan54** [comment]
@bwp91 definitely, let me look at the dependant projects to see about usage etc

One challenge with changing a unique ID definition is that all the devices will appear as new on implementation.  Which will cause issues for the dependant projects.

But I recall over riding this, so it may be a non issue.

**Comment by NorthernMan54** [comment]
Checking dependencies, only UI, GSH and Homebridge-automation use this package

Homebridge-automation overwrites the supplied unique ID - https://github.com/NorthernMan54/node-red-contrib-homebridge-automation/blob/f5bf77077439179325bbf01367f8c4dcd8f23e57/src/hbConfigNode.js#L70

GSH - And it is overwritten here as well, but with the broken approach - https://github.com/homebridge-plugins/homebridge-gsh/blob/f499173ed59952e274d773a5c01d9651ac8ea0f9/src/hap.ts#L381

When I refreshed GSH, I did not change the uniqueID logic, as it would break the setup of all existing users.....

So we can go ahead with this, and I will do a test of the packages when it is in Beta
